pymoveit2/moveit2.py

```diff
@@ -1148,6 +1153,59 @@ def allowed_planning_time(self, value: float):
 
         self.__move_action_goal.request.allowed_planning_time = value
 
+    def add_collision_mesh(
+        self,
+        id: str,
+        position: Union[Point, Tuple[float, float, float]],
+        quat_xyzw: Union[Quaternion, Tuple[float, float, float, float]],
+        filename: str,
+        frame_id: Optional[str] = None,
+        operation: int = CollisionObject.ADD,
+    ):
+
+        mesh = trimesh.load(filename)
+        msg = CollisionObject()
+        msg.id = id
+        msg.operation = operation
+        pose = Pose()
+        pose.position.x = float(position[0])
+        pose.position.y = float(position[1])
+        pose.position.z = float(position[2])
+        pose.orientation.x = float(quat_xyzw[0])
+        pose.orientation.y = float(quat_xyzw[1])
+        pose.orientation.z = float(quat_xyzw[2])
+        pose.orientation.w = float(quat_xyzw[3])
+        msg.pose = pose
+
+        msg.mesh_poses.append(Pose()) # Only global poses
+
+        msg.meshes.append(Mesh())
+        for face in mesh.faces:
+            tri = MeshTriangle()
+            for i in range(3):
+                tri.vertex_indices[i] = face[i]
+            msg.meshes[-1].triangles.append(tri)
+        for vert in mesh.vertices:
+            point = Point()
+            point.x = vert[0]
+            point.y = vert[1]
+            point.z = vert[2]
+            msg.meshes[-1].vertices.append(point)
```

Comment:
I think this might be more efficient.

```suggestion
        mesh_msg = Mesh(
            triangles=[MeshTriangle(vertex_indices=face) for face in mesh.faces],
            vertices=[Point(x=vert[0], y=vert[1], z=vert[2]) for vert in mesh.vertices],
        )
        msg.meshes.append(mesh_msg)
```